src/coreclr/debug/createdump/specialthreadinfo.h

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// ******************************************************************************
+// WARNING!!!: This code is also used by SOS in the diagnostics repo. Should be
+// updated in a backwards and forwards compatible way.
+// See: https://github.com/dotnet/diagnostics/blob/main/src/SOS/inc/specialthreadinfo.h
+// ******************************************************************************
+
+// This defines a workaround to the MacOS dump format not having the OS process
+// and thread ids that SOS needs to map thread "indexes" to thread "ids". The MacOS
+// createdump adds this special memory region at this specific address that is not
+// in the user or kernel address spaces. lldb is find with it.
+
+#define SPECIAL_THREADINFO_SIGNATURE "THREADINFO"
+
+const uint64_t SpecialThreadInfoAddress = 0x7fffffff00000000;
+
+struct SpecialThreadInfoHeader
+{
+    char signature[16];
+    uint32_t pid;
```

Comment:
I wonder if I should add the number of threads here (the number of SpecialThreadInfoEntry's) to make sure all the threads make it "through" lldb.  In createdump it would be always and exactly the number of thread_commands written to the dump so I didn't think it is necessary.  I haven't seen any case yet where lldb does something to the number or order of threads, but adding it would allow the SOS plugin code that is parsing it to add yet another check.

Reply:
Done.